unit-tests/unit-test-long.cpp

```diff
@@ -204,3 +205,210 @@ TEST_CASE("multicam_streaming", "[live][multicam]")
+            std::vector<rs2::stream_profile> modes;
+            REQUIRE_NOTHROW(modes = subdevice.get_stream_profiles());
+            REQUIRE(modes.size() > 0);
+            for (auto profile : modes)
+            {
+                if (max_fps < profile.fps())
+                {
+                    max_fps = profile.fps();
+                }
+            }
+        }
+
+        for (auto && subdevice : list) {
+            if (!subdevice.supports(RS2_OPTION_GLOBAL_TIME_ENABLED))
+                continue;
+            std::vector<rs2::stream_profile> modes;
+            REQUIRE_NOTHROW(modes = subdevice.get_stream_profiles());
+
+            REQUIRE(modes.size() > 0);
+            CAPTURE(subdevice.get_info(RS2_CAMERA_INFO_NAME));
+
+            //the test will be done only on the profile with maximal fps:
+            for (auto profile : modes)
+            {
+                if (profile.fps() < max_fps)
+                    continue;
+
+                CAPTURE(profile.format());
+                CAPTURE(profile.fps());
+                CAPTURE(profile.stream_type());
+                CAPTURE(profile.stream_index());
+                if (auto video = profile.as<video_stream_profile>())
+                {
+                    CAPTURE(video.width());
+                    CAPTURE(video.height());
+                }
+
+                double max_diff_system_global_time;
+                std::vector<double> all_results_gts_on;
+                std::vector<double> all_results_gts_off;
+                const int num_of_runs(7);
+                for (int i = 0; i < 7; i++)
+                {
+                    run_sensor(subdevice, profile, true, i, max_diff_system_global_time);
+                    all_results_gts_on.push_back(max_diff_system_global_time);
+                    std::cout << "Ran iteration " << i << "/" << num_of_runs << " - gts-ON: max_diff_system_global_time=" << max_diff_system_global_time << std::endl;
+
+                    run_sensor(subdevice, profile, false, i, max_diff_system_global_time);
+                    all_results_gts_off.push_back(max_diff_system_global_time);
+                    std::cout << "Ran iteration " << i << "/" << num_of_runs << " - gts-OFF: max_diff_system_global_time=" << max_diff_system_global_time << std::endl;
+                }
+                std::nth_element(all_results_gts_on.begin(), all_results_gts_on.begin() + all_results_gts_on.size() / 2, all_results_gts_on.end());
+                double median_diff_gts_on = all_results_gts_on[all_results_gts_on.size() / 2];
+                std::nth_element(all_results_gts_off.begin(), all_results_gts_off.begin() + all_results_gts_off.size() / 2, all_results_gts_off.end());
+                double median_diff_gts_off = all_results_gts_off[all_results_gts_off.size() / 2];
+                CAPTURE(median_diff_gts_on);
+                CAPTURE(median_diff_gts_off);
+
+                REQUIRE(median_diff_gts_on > 0.5*median_diff_gts_off);
```

Comment:
During 10 second Gyto will generate 4000 frames. Imo the median will not suffice.